src/junox/lfo.js

```diff
@@ -1,18 +1,46 @@
 export default class LFO {
-  constructor ({ frequency, sampleRate }) {
-    this.frequency = frequency
-    this.sampleRate = sampleRate
+  constructor ({ frequency, delay = 0, sampleRate }) {
+    this.anchorRate = 1 / sampleRate
     this.phase = 0
-    this.tick()
+    this.setRate(frequency)
+    this.setDelay(delay)
+    this.trigger()
   }
 
-  tick () {
-    this.samplesPerPeriod = Math.round(this.sampleRate / this.frequency)
-    this.phase = (this.phase + 1) % this.samplesPerPeriod
+  setRate (frequency) {
+    this.incrementRate = this.anchorRate * frequency
+  }
+
+  setDelay (delay) {
+    this.delay = delay * this.anchorRate
+  }
+
+  trigger () {
+    this.delayPhase = 0
+  }
+
+  delayEnv () {
+    this.delayPhase += this.delay
+    if (this.delay < 0.0001 || this.delayPhase > 2) {
+      return 1
+    } else if (this.delayPhase < 1) {
+      return 0
+    }
+    // from juno manual the volume seems to grows by the same distance it's delayed
+    return this.delayPhase - 1
   }
 
   render () {
-    const phase = this.phase / this.samplesPerPeriod
-    return (2.0 * Math.sin(Math.sin(2.0 * Math.PI * phase))) / Math.PI
+    let out = 0
+    this.phase += this.incrementRate
+    if (this.phase > 1.01) {
```

Comment:
ooooh!  Just found this feature on GitHub.

incrementRate != phase.

We are emulating a triangle wave, so I have taken certain liberties with the implementation.  It starts from zero, then using "+incrementRate" each sample-step until it reaches a total of +1.  Then I reverse the direction, and use "-incrementRate" each step until I reach -1, then it returns to the positive direction.  So the output is already triangular without needing to apply the code you have at line 39.